### PR TITLE
cmd/libsnap-confine-private: add sc_cleanup_deep_strv

### PR DESCRIPTION
**Pull request by Meulengracht**
Deep cleanup of string arrays. Utility function needed for PR https://github.com/snapcore/snapd/pull/12288.

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
This looks fine, thank you

**Comment by Meulengracht** [comment]
I just realized due to the build failing, I am not allowed to introduce this function as it's not used. I guess I'll have to include it with the PR that needs it. It was just an easy pick for separation.

File: cmd/libsnap-confine-private/cleanup-funcs-test.c [hunk 1]
```diff
@@ -142,6 +142,31 @@ static void test_cleanup_close(void)
 	g_assert_cmpint(fd, ==, -1);
 }
 
+static void test_cleanup_deep_strv(void)
```
Comment (valentindavid) [inline cmd/libsnap-confine-private/cleanup-funcs-test.c:145]: Could you register that test later in `init()`?

**Review by valentindavid — CHANGES_REQUESTED** [review]
Looks fine. Tested with valgrind. But I had to add the test since you forgot it. Could you add it? See the comment.

**Review by valentindavid — APPROVED** [review]

**Pull request merged**